pulp_ansible/app/tasks/collections.py

```diff
@@ -461,6 +461,11 @@ def is_new_highest(new, old):
 
     # exit if the new CV is not higher
     if not is_new_highest(Version(collection_version.version), Version(last_highest.version)):
+        # Ensure that this collection_version doesn't have is_highest improperly set
+        if collection_version.is_highest:
+            collection_version.is_highest = False
+            if save:
```

Comment:
What is this save trying to tell us?

Reply:
What do you mean? I explained the bug in the issue. This `if save` is to just be consistent with how the function can be called.

Reply:
That previous condition is for an odd scenario where none of the collection versions for a collection had `is_highest` set. The reason why `save` is conditional is for an optimization since this function is called to update a collection version before it is saved in the `CollectionSaver` stage or the `import_collection` task, there is no point in saving the collection twice in a row.